src/middlewared/middlewared/utils/metrics/meminfo.py

```diff
@@ -0,0 +1,22 @@
+import humanfriendly
+from dataclasses import dataclass
+
+
+@dataclass(slots=True, frozen=True, kw_only=True)
+class MemoryInfo:
+    total: int
+
+
+def get_memory_info() -> MemoryInfo:
+    with open('/proc/meminfo') as f:
+        meminfo = {
+            s[0]: humanfriendly.parse_size(s[1], binary=True)
+            for s in [
+                line.split(':', 1)
+                for line in f.readlines()
```

Comment:
Please change this to what I suggested. This reading the entire file into memory just to parse the first line.